api/v1alpha3/machine_types.go

```diff
@@ -89,6 +89,12 @@ type MachineSpec struct {
 	// Must match a key in the FailureDomains map stored on the cluster object.
 	// +optional
 	FailureDomain *string `json:"failureDomain,omitempty"`
+
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// +optional
+	NodeDrainTimeout int64 `json:"nodeDrainTimeout,omitempty"`
```

Comment:
I wonder if it would make more sense to add this to `MachineDeploymentStrategy` instead (and then to KCP separately) https://github.com/CecileRobertMichon/cluster-api/blob/2b71eebdec7e361e9baeb8f1454f8a6e0268c169/api/v1alpha3/machinedeployment_types.go#L101

What's the reasoning for setting this value for each machine individually?

Reply:
Hi @CecileRobertMichon, thanks for your review.
We actually don't set this value for each machine individually, but setting in the `machineSpec` of  `machinedeployment`. 
For example: 

```
apiVersion: cluster.x-k8s.io/v1alpha3
kind: MachineDeployment
metadata:
...
spec:
...
  template:
    metadata:
      labels:
        cluster.x-k8s.io/cluster-name: my-cluster
        nodepool: nodepool-0
    spec:
      clusterName: my-cluster
      version: v1.15.3
      bootstrap:
        ...
      infrastructureRef:
        ...
      nodeDrainTimeout: 180 

```

Reply:
Right, but it's still part of the template instead of being part of the MachineDeployment spec directly

Reply:
Draining is a per-node (machine) operation, though. Also look at the Pod spec - it has e.g. `terminationGracePeriodSeconds`, which is a similar type of thing to nodeDrainTimeout.

Reply:
@CecileRobertMichon ok to mark resolved?

Reply:
yes, marking as resolved 👍